app/plugin_api.go

```diff
@@ -361,6 +362,29 @@ func (api *PluginAPI) GetFileInfo(fileId string) (*model.FileInfo, *model.AppErr
 	return api.app.GetFileInfo(fileId)
 }
 
+func (api *PluginAPI) GetFilePreview(fileId string) ([]byte, *model.AppError) {
+	info, err := api.app.GetFileInfo(fileId)
+	if err != nil {
+		return nil, err
+	}
+
+	if info.PreviewPath == "" {
+		return nil, model.NewAppError("GetFilePreview", "plugin_api.get_file_preview.no_preview.app_error", nil, "file_id="+info.Id, http.StatusBadRequest)
+	}
+
+	fileReader, err := api.app.FileReader(info.PreviewPath)
```

Comment:
If we're just reading it into a buffer to return, can we just use `api.app.ReadFile(path)` like on line 389?

@crspeller, should we be looking into more complex implementations here that involve passing a reader to the plugin instead, to help with larger files?